src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2543,7 +2555,8 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             Stream sourceLinkStream,
             IEnumerable<EmbeddedText> embeddedTexts,
             CompilationTestData testData,
-            CancellationToken cancellationToken)
+            CancellationToken cancellationToken,
+            bool filterDiagnostics = true)
```

Comment:
nit: it seems this method is only called from a handful of locations, could we just pass the value explicitly and make cancellationtoken the last parameter instead?